Update GH to use shared workflow

chris1984 opened the pull request.

ofedoren commented on .github/workflows/ci.yml:
```diff
@@ -1,53 +1,20 @@
-      run: bundle exec rake test:foreman_virt_who_configure
+  rubocop:
+    name: Rubocop
+    uses: theforeman/actions/.github/workflows/rubocop.yml@v0
```
We can change this to
```suggestion
    uses: theforeman/actions/.github/workflows/rubocop.yml@v0
    with:
      command: bundle exec rake foreman_virt_who_configure:rubocop
```
or

```suggestion
    uses: theforeman/actions/.github/workflows/rubocop.yml@v0
    with:
      command: bundle exec rubocop --parallel --format github
```

ofedoren reviewed (commented):
@chris1984, let's unblock this? :)

chris1984 commented:
> @chris1984, let's unblock this? :)

Yeah I want to get this in

ofedoren commented:
Seems like CI can't find this task:
```
rake aborted!
Don't know how to build task 'foreman_virt_who_configure:rubocop' (See the list of available tasks with `rake --tasks`)
```

I can't seem to find this task locally either. But the previous GHA definition was using that tasks for rubocop. No idea what happened.

Also, I've noticed this uses pretty weird rubocop definition: https://github.com/theforeman/foreman_virt_who_configure/blob/master/.rubocop.yml#L3

We should probably remove that and either create own `rubocop_todo.yml` here or incorporate https://github.com/theforeman/theforeman-rubocop to be consistent.

As a workaround, for this PR we could go with second way from https://github.com/theforeman/foreman_virt_who_configure/pull/186#discussion_r1469691883.

chris1984 commented:
> Seems like CI can't find this task:
> 
> ```
> rake aborted!
> Don't know how to build task 'foreman_virt_who_configure:rubocop' (See the list of available tasks with `rake --tasks`)
> ```
> 
> I can't seem to find this task locally either. But the previous GHA definition was using that tasks for rubocop. No idea what happened.
> 
> Also, I've noticed this uses pretty weird rubocop definition: https://github.com/theforeman/foreman_virt_who_configure/blob/master/.rubocop.yml#L3
> 
> We should probably remove that and either create own `rubocop_todo.yml` here or incorporate https://github.com/theforeman/theforeman-rubocop to be consistent.
> 
> As a workaround, for this PR we could go with second way from [#186 (comment)](https://github.com/theforeman/foreman_virt_who_configure/pull/186#discussion_r1469691883).

Sounds good, fixed it up and rebased, lets see if it goes through

chris1984 commented:
Well that got further, but it didn't like the gem, guess maybe we should just use local rubocop?

ofedoren commented:
@chris1984, well, it didn't like the gem because it doesn't get installed since there is no dependency on it.

Could you please try to add this into https://github.com/theforeman/foreman_virt_who_configure/blob/master/foreman_virt_who_configure.gemspec#L31
as the dev dependency instead of existing rubocop?

```ruby
s.add_development_dependency 'theforeman-rubocop'
```

ofedoren commented:
Actually, I guess, it'll be better to do something like https://github.com/theforeman/foreman_kubevirt/commit/d4fe841eca885f79367516676f27eddb4fd0b9d0

chris1984 commented:
> Actually, I guess, it'll be better to do something like [theforeman/foreman_kubevirt@d4fe841](https://github.com/theforeman/foreman_kubevirt/commit/d4fe841eca885f79367516676f27eddb4fd0b9d0)

Sounds good let me go this way

chris1984 commented:
> Actually, I guess, it'll be better to do something like [theforeman/foreman_kubevirt@d4fe841](https://github.com/theforeman/foreman_kubevirt/commit/d4fe841eca885f79367516676f27eddb4fd0b9d0)

I wonder if the make-mo messages are causing the failure or that github formatter

ofedoren commented on .github/workflows/ci.yml:
```diff
@@ -1,53 +1,22 @@
+    name: Rubocop
+    uses: theforeman/actions/.github/workflows/rubocop.yml@v0
+    with:
+      command: bundle exec rubocop --parallel --format github
```
```suggestion
      command: bundle exec rubocop
```

ofedoren commented:
Let's remove formatting for now.

chris1984 commented:
Fingers crossed 🤞

chris1984 commented:
That looks better let me fix up the ruby lint error, at least it ran

chris1984 commented:
Well Rubcop is not really inspecting anything lol but at least it moved onto the other part

ofedoren commented on .github/workflows/ci.yml:
```diff
@@ -1,53 +1,22 @@
+    name: Ruby
+    needs: rubocop
+    uses: theforeman/actions/.github/workflows/foreman_plugin.yml@v0
+    with:
+      plugin: foreman_virt_who_configure
```
```suggestion
    with:
      plugin: foreman_virt_who_configurewith:
      postgresql_container: ghcr.io/theforeman/postgresql-evr
      test_existing_database: false
```

ofedoren reviewed (approved):
Seems legit to me, but I'm no @evgeni :/

The suggestions were based on https://github.com/Katello/katello/pull/10780/files

evgeni commented on .github/workflows/ci.yml:
```diff
@@ -1,53 +1,24 @@
-  run:
-    working-directory: ./foreman
+
+on: pull_request
```
please keep `push` in that list, otherwise you don't get a cache of gems

evgeni reviewed (approved):
one tiny thing, but it'd work either way, so ack

evgeni commented:
> Seems legit to me, but I'm no @evgeni :/

you don't want to be, trust me :)

chris1984 commented:
@ofedoren should we just take out rubocop then since it's only looking at the rake file?

ofedoren commented:
> @ofedoren should we just take out rubocop then since it's only looking at the rake file?

@chris1984, I don't have any opinion regarding rubocop :/ If you want this in this repo, then it should be probably fixed to run across all the files (it's better to do in a separate PR).

chris1984 commented:
@ofedoren should we just take out rubocop then since it's only looking at the rake file?

> > @ofedoren should we just take out rubocop then since it's only looking at the rake file?
> 
> @chris1984, I don't have any opinion regarding rubocop :/ If you want this in this repo, then it should be probably fixed to run across all the files (it's better to do in a separate PR).

Sounds good, once this goes green then we can merge and I can worry about rubocop
